org.eclipse.winery.lsp/src/main/java/org/eclipse/winery/lsp/Server/ServerCore/Completion/CompletionItemGetter.java

```diff
@@ -0,0 +1,67 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache Software License 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ *******************************************************************************/
+package org.eclipse.winery.lsp.Server.ServerCore.Completion;
+
+import org.eclipse.lsp4j.*;
+import org.eclipse.lsp4j.jsonrpc.messages.Either;
+import org.eclipse.winery.lsp.Server.ServerAPI.API.context.LSContext;
+import org.eclipse.winery.lsp.Server.ServerCore.DataModels.ArtifactType;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class CompletionItemGetter {
+    public List<CompletionItem> getAvailableArtifactTypes(LSContext lsContext) {
+        List<String> artifactTypes = new ArrayList<>();
+        for (Map.Entry<String, ArtifactType> itr:lsContext.getToscaFile().artifactTypes().get().entrySet()) {
+            artifactTypes.add(" " + itr.getKey());
+        }
+        return artifactTypes.stream()
+            .map(type -> {
+                CompletionItem item = new CompletionItem(type);
+                item.setKind(CompletionItemKind.Value);
+                return item;
+            })
+            .collect(Collectors.toList());
+    }
+
+    public List<CompletionItem> getTOSCAFileKeywords(Position position) {
+        List<String> keywords = List.of(
+            "tosca_definitions_version:", "description:", "metadata:", "dsl_definitions:",
+            "artifact_types:", "data_types:", "capability_types:", "interface_types:",
+            "relationship_types:", "node_types:", "group_types:", "policy_types:",
+            "repositories:", "functions:", "profile:", "imports:", "service_template:"
```

Comment:
This reads not context dependent. For instance, I cannot complete "tosa_definitions_version" when I am in a servicete_template.

(You can mark areas which are work in progress also with "TODO")

Reply:
Yeah sure I agree.